### PR TITLE
Update personal keymap

### PR DESCRIPTION
**Pull request by rpbaptist**
## Description

This is an update of my personal keymap. Current keymap in master branch is broken and outdated.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/crkbd/keymaps/rpbaptist/config.h [hunk 1]
```diff
@@ -20,115 +20,82 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-#ifdef OLED_ENABLE
-#    define OLED_TIMEOUT 600000
-#endif
-
-// #define FORCE_NKRO
-
-#define QMK_ESC_OUTPUT F4  // usually COL
-#define QMK_ESC_INPUT D4   // usually ROW
-#define QMK_LED B0
+#define USB_POLLING_INTERVAL_MS 1
```
Comment (drashna) [inline keyboards/crkbd/keymaps/rpbaptist/config.h:26]: This is unneeded, as it is the default setting.
```suggestion
```

File: keyboards/crkbd/keymaps/rpbaptist/oled.c
Comment (drashna) [inline keyboards/crkbd/keymaps/rpbaptist/oled.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/crkbd/keymaps/rpbaptist/oled.h
Comment (drashna) [inline keyboards/crkbd/keymaps/rpbaptist/oled.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/crkbd/keymaps/rpbaptist/rgb.c
Comment (drashna) [inline keyboards/crkbd/keymaps/rpbaptist/rgb.c:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/crkbd/keymaps/rpbaptist/rgb.c [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+void matrix_scan_rgb(void) {
+    if (user_config.rgb_matrix_idle_anim && rgb_matrix_get_mode() == user_config.rgb_matrix_active_mode && sync_timer_elapsed32(idle_timer) > user_config.rgb_matrix_idle_timeout) {
+        if (user_config.rgb_layer_indicator) {
+            rgb_matrix_turn_off_underglow();
+        }
+        rgb_matrix_update_current_mode(user_config.rgb_matrix_idle_mode);
+    }
+}
+
+void matrix_scan_user(void) {
```
Comment (drashna) [inline keyboards/crkbd/keymaps/rpbaptist/rgb.c:249]: should use:
```suggestion
void housekeeping_task_user(void) {
```

File: keyboards/crkbd/keymaps/rpbaptist/rgb.h
Comment (drashna) [inline keyboards/crkbd/keymaps/rpbaptist/rgb.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

File: keyboards/crkbd/keymaps/rpbaptist/rgb.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        bool     rgb_matrix_idle_anim : 1;
+        uint8_t  rgb_matrix_active_mode : 4;
+        uint8_t  rgb_matrix_idle_mode : 4;
+        uint8_t  rgb_matrix_active_speed : 8;
+        uint8_t  rgb_matrix_idle_speed : 8;
+        uint32_t rgb_matrix_idle_timeout : 32;
+    };
+} user_config_t;
+
+user_config_t user_config;
```
Comment (drashna) [inline keyboards/crkbd/keymaps/rpbaptist/rgb.h:18]: This should be an extern, and declared in a c file.
```suggestion
extern user_config_t user_config;
```

Also, you may want: 
```c
_Static_assert(sizeof(user_config_t) == sizeof(uint32_t), "Userspace EECONFIG out of spec.");
```

File: keyboards/crkbd/keymaps/rpbaptist/rpbaptist.h
Comment (drashna) [inline keyboards/crkbd/keymaps/rpbaptist/rpbaptist.h:1]: Could you add a GPL2+ compatible license header here?  All of the source files need to have a license explicitly listed.

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/crkbd/keymaps/rpbaptist/rpbaptist.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
+#pragma once
+#include QMK_KEYBOARD_H
+
+#if defined(RGB_MATRIX_ENABLE) || defined(OLED_ENABLE)
+uint32_t idle_timer;
```
[the remark this answers is not in the document]
Comment (rpbaptist) [reply keyboards/crkbd/keymaps/rpbaptist/rpbaptist.h:5]: I was not aware of that function. I'll happily use that instead. Thank you!

File: keyboards/crkbd/keymaps/rpbaptist/glcdfont_multi.c [hunk 5]
```diff
@@ -0,0 +1,54 @@
+#include "progmem.h"
+
+// Helidox 8x6 font with QMK Firmware Logo
+// Online editor: https://joric.github.io/qle/
+
+#if defined(THEME_GODSPEED) || defined(THEME_GODSPEED_ARES)
```
[the remark this answers is not in the document]
Comment (rpbaptist) [reply keyboards/crkbd/keymaps/rpbaptist/glcdfont_multi.c:6]: Will do!

**Comment by rpbaptist** [comment]
I moved the keymap to the `rev1` directory as the build was too large for the `r2g` version.

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**